File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/util/Mappers.java
package com.igorganapolsky.vibratingwatchapp.util;

import com.igorganapolsky.vibratingwatchapp.domain.repo.TimerEntity;
import com.igorganapolsky.vibratingwatchapp.domain.model.VibrationModel;
import com.igorganapolsky.vibratingwatchapp.domain.model.TimerModel;

public class Mappers {

    public static TimerModel mapToTimerModel(TimerEntity entity) {
        TimerModel model = new TimerModel();
        model.setId(entity.getId());
        model.setRepeat(entity.getRepeat());
        model.setVibrationCount(entity.getBuzzCount());
        model.setType(VibrationModel.Type.valueOf(entity.getBuzzType()));
        model.setState(TimerModel.State.valueOf(entity.getState()));
        model.setVibrationTimeInSecs(entity.getBuzzTime());
        model.setHours(TimerTransform.getHours(entity.getMilliseconds()));
        model.setMinutes(TimerTransform.getMinutes(entity.getMilliseconds()));
        model.setSeconds(TimerTransform.getSeconds(entity.getMilliseconds()));
        return model;
    }

    public static TimerEntity mapToTimerEntity(TimerModel model) {
        TimerEntity entity = new TimerEntity();
        entity.setRepeat(model.getRepeat());
        entity.setState(TimerModel.State.FINISHED.name());
        entity.setBuzzCount(model.getVibrationCount());
        entity.setBuzzType(model.getType().name());
        entity.setBuzzTime(model.getVibrationTimeInSecs());
        entity.setMilliseconds(model.getTimeInMillis());
        return entity;
    }

    public static VibrationModel mapToBuzzSetup(TimerModel model) {
        return new VibrationModel(model.getType(), model.getVibrationCount(), model.getVibrationTimeInSecs());
    }
}

File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/core/Mappers.kt
package com.igorganapolsky.vibratingwatchapp.core

import com.igorganapolsky.vibratingwatchapp.data.TimerEntity
import com.igorganapolsky.vibratingwatchapp.data.VibrationModel
import com.igorganapolsky.vibratingwatchapp.domain.model.TimerModel

object Mappers {

    fun mapToTimerModel(entity: TimerEntity): TimerModel {
        val model = TimerModel()
        model.id = entity.id
        model.repeat = entity.repeat
        model.vibrationCount = entity.buzzCount
        model.type = VibrationModel.Type.valueOf(entity.buzzType)
        model.state = TimerModel.State.valueOf(entity.state)
        model.vibrationTimeInSecs = entity.buzzTime
        model.hours = TimerTransform.getHours(entity.milliseconds)
        model.minutes = TimerTransform.getMinutes(entity.milliseconds)
        model.seconds = TimerTransform.getSeconds(entity.milliseconds)
        return model
    }

    fun mapToTimerEntity(model: TimerModel): TimerEntity {
        val entity = TimerEntity()
        entity.repeat = model.repeat
        entity.state = TimerModel.State.FINISHED.name
        entity.buzzCount = model.vibrationCount
        entity.buzzType = model.type!!.name
        entity.buzzTime = model.vibrationTimeInSecs
        entity.milliseconds = model.timeInMillis
        return entity
    }

    fun mapToBuzzSetup(model: TimerModel): VibrationModel {
        return VibrationModel(
            model.type!!,
            model.vibrationCount,
            model.vibrationTimeInSecs
        )
    }
}
